src/main/java/ro/msg/learning/shop/service/impl/CustomerServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package ro.msg.learning.shop.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.entity.Customer;
+import ro.msg.learning.shop.entity.Product;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.service.CustomerService;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class CustomerServiceImpl implements CustomerService {
+
+    @Autowired
```

Comment:
You should better use constructor injection here. Sometimes field injection is ambiguous, while constructor-based injection enforces the dependency requirement. It has more advantages than field injection. You can do this with Lombok's **@RequiredArgsConstructor** annotation. Please adapt this wherever this situation occurs.
https://mohamed-kashif.medium.com/constructor-injection-vs-field-injection-in-java-2265200fe203
https://www.baeldung.com/spring-injection-lombok